### ISSUE TITLE
DevModeInitializer.getBaseDirectoryFallback() fails under Windows

### ISSUE THREAD

**Issue by SonReimer** [vaadin/flow#16605]
### Description of the bug

Starting an application from eclipse under Windows leads to a failure in code committed with
https://github.com/vaadin/flow/pull/16266.
In our case the newly introduced code
```
            URL url = DevModeInitializer.class.getClassLoader()
                    .getResource(".");
            if (url != null && url.getProtocol().equals("file")) {
                // URI decodes the path so that e.g. " " works correctly
                String path = url.toURI().getPath();
                if (path.endsWith("/target/classes/")) {
                    return path.replaceFirst("/target/classes/$", "");
                }
            }
```
returns the following path: `/C:/projects/platform-gear-13/runtime/refimpl/frontend`
which then can not be converted in DevModeUsageStatistic:92 :
`String projectId = ProjectHelpers.generateProjectId(projectFolder);`
because it cannot be converted by
`Paths.getPath(String)`
under Windows.

The exception is
`Caused by: java.nio.file.InvalidPathException: Illegal char <:> at index 2: /C:/projects/platform-gear-13/runtime/refimpl/frontend`

### Expected behavior

It should work.

### Minimal reproducible example

Start an application in eclispe under Windows

### Versions

- Vaadin / Flow version: 23.3.11/23.37
- Java version: 11
- OS version: Windows
- Browser version (if applicable): not relevant
- Application Server (if applicable): not relevant
- IDE (if applicable): Eclipse

**Comment by Artur-** [comment]
I wonder if `Path.of(uri)` i.e. `Path.of(url.toURI())` would be the correct way to do this..

**Comment by mcollovati** [comment]
> I wonder if `Path.of(uri)` i.e. `Path.of(url.toURI())` would be the correct way to do this..

It seems like a good solution :+1:

**Comment by Artur-** [comment]
Slightly different fix in #16611
